src/Codeception/Module/Doctrine2.php

```diff
@@ -401,7 +401,15 @@ public function haveFakeRepository(string $className, array $methods = []): void
 
             $reflectedRepositoryFactory = new ReflectionClass($repositoryFactory);
 
-            if ($reflectedRepositoryFactory->hasProperty('repositoryList')) {
+            // Do not call $reflectedRepositoryFactory->isReadOnly() directly because
+            // phpstan will complain about a non-existing method when using PHP 8.0.
+            // isReadOnly() is available as-of PHP 8.1.
+            if ($reflectedRepositoryFactory->hasMethod('isReadOnly') && $reflectedRepositoryFactory->getMethod('isReadOnly')->invoke(null)) {
```

Comment:
[`EntityManager::$repositoryFactory`](https://github.com/doctrine/orm/blob/5b8b5f28f535e1f03b54dcfb0427407ed92f5b72/src/EntityManager.php#L87) was made `readonly` in ORM v3.